File: src/main/java/org/example/gogoma/controller/request/CreateUserMarathonRequest.java
package org.example.gogoma.controller.request;

import lombok.*;
import org.example.gogoma.domain.usermarathon.enums.PaymentType;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreateUserMarathonRequest {

    private int marathonId;

    private String address;

    private PaymentType paymentType;

    private String paymentAmount;

    private LocalDateTime paymentDateTime;

    private String courseType;
}

File: app/src/main/java/com/example/gogoma/data/model/CreateUserMarathonRequest.kt
package com.example.gogoma.data.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class CreateUserMarathonRequest(
    val marathonId: Int,
    val address: String,
    val paymentType: PaymentType, // KAKAO_PAY, BANK_DEPOSIT
    val paymentAmount: String,
    val paymentDateTime: String,
    val courseType: Int
) : Parcelable
